metals/src/main/scala/scala/meta/internal/metals/ScalafmtConfig.scala

```diff
@@ -0,0 +1,206 @@
+package scala.meta.internal.metals
+
+import java.nio.file.FileSystems
+
+import scala.collection.JavaConverters._
+import scala.util.Try
+
+import scala.meta.io.AbsolutePath
+
+import com.typesafe.config.Config
+import com.typesafe.config.ConfigFactory
+import com.typesafe.config.ConfigParseOptions
+import com.typesafe.config.ConfigSyntax
+import com.typesafe.config.parser.ConfigDocument
+import com.typesafe.config.parser.ConfigDocumentFactory
+
+object ScalafmtConfig {
+
+  sealed trait SyntaxError
+  case object SyntaxError extends SyntaxError
+
+  case class FileOverride(
+      glob: String,
+      dialect: ScalafmtDialect
+  )
+
+  sealed trait PathMatcher {
+    def matches(path: AbsolutePath): Boolean
+  }
+  object PathMatcher {
+    final class Nio(pattern: String) extends PathMatcher {
+      private val matcher = FileSystems.getDefault().getPathMatcher(pattern)
+      def matches(path: AbsolutePath): Boolean = matcher.matches(path.toNIO)
+    }
+
+    final class Regex(regex: String) extends PathMatcher {
+      private val pattern = java.util.regex.Pattern.compile(regex)
+      def matches(path: AbsolutePath): Boolean =
+        pattern.matcher(path.toString).find()
+    }
+  }
+
+  case class ConfigValues(
+      version: Option[String],
+      runnerDialect: Option[ScalafmtDialect],
+      fileOverrides: Map[String, ScalafmtDialect],
+      includeFilters: List[PathMatcher],
+      excludeFilters: List[PathMatcher]
+  ) {
+
+    def overrideFor(path: AbsolutePath): Option[ScalafmtDialect] = {
+      val fs = FileSystems.getDefault
```

Comment:
```suggestion
      val fileSystem = FileSystems.getDefault
```
it's easier to understand. Reviewers need to think about what fs stands for. This is obviously a nitpick, but I am kind of inspired by the recent keynote at Scalacon about reading code.